smolyay/basis.py

```diff
@@ -429,3 +445,70 @@ def __init__(self, basis_functions):
     def basis_functions(self):
         """list: Basis functions."""
         return self._basis_functions
+    
+    @property
```

Comment:
I also don't think this needs to be in the public API. Surrogates that need it can do this themselves since you made this a sequence now:
```py
if any(bf._is_complex for bf in basis_set):
```